plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -30,6 +30,12 @@ import esb from 'elastic-builder';
 import { isEmpty, isNaN as nan, isNumber } from 'lodash';
 import { Logger } from 'winston';
 
+export type ElasticSearchClientOptions = ConstructorParameters<
+  typeof Client
+>[0] & {
+  provider?: 'aws' | 'elastic' | 'generic';
```

Comment:
Feels a little odd to mix `provider` (something specific to us) with the rest of the client options on the same level.

Might be safer to make them siblings of one another, rather than unioning them as types?

```typescript
type ElasticSearchClientOptions = {
  provider?: 'aws' | 'elastic';
  options: ConstructorParameters<typeof Client>[0],
};
```

Reply:
But also, per your notes:

> _If we're not comfortable with coupling the parameter type of options to the version of @elastic/elasticsearch we use, then an alternative is to maintain our own separate configuration schema which is a subset of the elastic search ClientOptions..._

You'll likely need to run `yarn build:api-reports` at the root of this monorepo to get these builds passing, and in the diff of `api-report.md` for this plugin, there will likely be a line like `import { Client } from '@elastic/elasticsearch';` which will make the inclusion of the upstream type much more clearly a part of our interface.

So with that in mind, it'd be awesome if you could split it into its own (conforming) type, rather than relying on it directly here.

Reply:
Ok, I brought in the type. It's pretty gnarly, but it does keep us completely decoupled from the Elastic Search API.